Add login and register pages

Mendess2526 opened the pull request:
## Register
route: `/register/:id`

Register receives the attendee's id and prompts them to input their username, email, password and password confirmation.

If the form's submission has any errors these are displayed to the user: [register.js:44](https://github.com/cesium/moonstone/blob/fe131a9507576243735d369c74f7ad00f61f04af/src/pages/register.js#L44)

## Login
route: `/login`

Login prompts the attendee to input their email and password.

If the form's submission has any errors these are displayed to the user: [login.js:33](https://github.com/cesium/moonstone/blob/fe131a9507576243735d369c74f7ad00f61f04af/src/pages/login.js#L33)

## Both

Both login and register save the session token in localStorage (see [login.js:26](https://github.com/cesium/moonstone/blob/fe131a9507576243735d369c74f7ad00f61f04af/src/pages/login.js#L26) and [register.js:37](https://github.com/cesium/moonstone/blob/fe131a9507576243735d369c74f7ad00f61f04af/src/pages/register.js#L37))

herulume commented on src/pages/index.js:
```diff
@@ -1,8 +1,49 @@
+          >
+            <Text>Side item</Text>
+          </Sidebar>
+        </Box> */}
```
Remove commented code.

Mendess2526 replied:
This code was not commented by me, should I still remove it?

herulume replied:
Yes.

herulume commented on src/pages/login.js:
```diff
@@ -0,0 +1,76 @@
+import {Button, FormField, Text, TextInput} from "grommet";
+import request from "superagent/lib/client";
+
+const api_endpoint = '';
```
We should move this to a .env (or the standard way in react).

herulume commented on src/pages/login.js:
```diff
@@ -0,0 +1,76 @@
+const api_endpoint = '';
+
+export default class Login extends Component {
+
```
Remove empty line.

Mendess2526 replied:
Why?

herulume commented on src/pages/login.js:
```diff
@@ -0,0 +1,76 @@
+    this.setState({password: event.target.value});
+  };
+
+  handle_login_response(res) {
```
Mix of snake case and camel case.

herulume commented on src/pages/login.js:
```diff
@@ -0,0 +1,76 @@
+  }
+
+  login = () => {
+    /* TODO fix CORS and test this */
```
Remove this and do test it please.

Mendess2526 replied:
As I said in the PR message, I can't fully test this because I'm having problems with `CORS`.
`axios` seemed to work for you but @apedrob suggested using `superagent`.
@ZePedroResende @0urobor0s

herulume replied:
That should be in the PR itself as you put it. Such comments shouldn't go to production.

herulume commented on src/pages/register.js:
```diff
@@ -0,0 +1,127 @@
+import {Text, Button, FormField, TextInput} from "grommet";
+import request from "superagent/lib/client";
+
+const api_endpoint = '';
```
Same.

herulume commented on src/pages/register.js:
```diff
@@ -0,0 +1,127 @@
+const api_endpoint = '';
+
+export default class Register extends Component {
+
```
Same.

Mendess2526 replied:
Why?

herulume replied:
Standard react I think

herulume commented on src/pages/register.js:
```diff
@@ -0,0 +1,127 @@
+    this.setState({password_confirmation: event.target.value});
+  };
+
+  handle_register_response(res) {
```
Same. 
We could refactor this since it's used elsewhere, no?

Mendess2526 replied:
It's such a small function and although it's repeated it's still too specific to refactor in my opinion.

herulume commented on src/pages/register.js:
```diff
@@ -0,0 +1,127 @@
+  }
+
+  register = () => {
+    /* TODO fix CORS and test this */
```
Same

herulume commented on src/pages/register.js:
```diff
@@ -0,0 +1,127 @@
+          password: this.state.password,
+          password_confirmation: this.state.password_confirmation
+      })
+      .on('error', e => console.log(e))
```
Shouldn't it be put on 'errors'?

herulume commented on src/containers/defaultContainer.js:
```diff
@@ -5,6 +5,7 @@ const api_key = ''
 const api_endpoint = ''
```
We should move this to a .env (or the standard way in react).

Mendess2526 commented:
I admit my point about the `.env` was not very sound, changed.

The comment could be more informative, but I don't think it's useless. I changed it and opened the issue.
